chat-plugins/mafia.js

```diff
@@ -47,59 +49,109 @@
-	if (!json) {
-		// Create file
-		FS(LOGS_FILE).writeSync('{"leaderboard": {}, "mvps": {}, "hosts": {}, "plays": {}}');
-	} else {
-		logs = JSON.parse(json);
+/**
+ *
+ * @param {string} name
+ */
+function readFile(name) {
+	try {
+		const json = FS(name).readIfExistsSync();
+		if (!json) {
+			FS(name).writeSync("{}");
+			return false;
+		}
+		return JSON.parse(json);
+	} catch (e) {
+		if (e.code !== 'ENOENT') throw e;
 	}
-	/** @type {MafiaLogSection[]} */
-	const tables = ['leaderboard', 'mvps', 'hosts', 'plays'];
-	for (const section of tables) {
-		// Check to see if we need to eliminate an old month's data.
-		const month = new Date().toLocaleString("en-us", {month: "numeric", year: "numeric"});
-		if (!logs[section]) logs[section] = {};
-		if (!logs[section][month]) logs[section][month] = {};
-		if (Object.keys(logs[section]).length >= 3) {
-			// eliminate the oldest month(s)
-			let keys = Object.keys(logs[section]).sort((aKey, bKey) => {
-				const a = aKey.split('/');
-				const b = bKey.split('/');
-				if (a[1] !== b[1]) {
-					// year
-					if (parseInt(a[1]) < parseInt(b[1])) return -1;
-					return 1;
-				}
-				// month
-				if (parseInt(a[0]) < parseInt(b[0])) return -1;
+}
+/**
+ *
+ * @param {string} name
+ * @param {object} data
+ */
+function writeFile(name, data) {
+	FS(name).writeUpdate(() => (
+		JSON.stringify(data)
+	));
+}
+// Load logs
+logs = readFile(LOGS_FILE);
+if (!logs) logs = {leaderboard: {}, mvps: {}, hosts: {}, plays: {}};
+/** @type {MafiaLogSection[]} */
+const tables = ['leaderboard', 'mvps', 'hosts', 'plays'];
+for (const section of tables) {
+	// Check to see if we need to eliminate an old month's data.
+	const month = new Date().toLocaleString("en-us", {month: "numeric", year: "numeric"});
+	if (!logs[section]) logs[section] = {};
+	if (!logs[section][month]) logs[section][month] = {};
+	if (Object.keys(logs[section]).length >= 3) {
+		// eliminate the oldest month(s)
+		let keys = Object.keys(logs[section]).sort((aKey, bKey) => {
+			const a = aKey.split('/');
+			const b = bKey.split('/');
+			if (a[1] !== b[1]) {
+				// year
+				if (parseInt(a[1]) < parseInt(b[1])) return -1;
 				return 1;
-			});
-			while (keys.length > 2) {
-				const curKey = keys.shift();
-				if (!curKey) break; // should never happen
-				delete logs[section][curKey];
 			}
+			// month
+			if (parseInt(a[0]) < parseInt(b[0])) return -1;
+			return 1;
+		});
+		while (keys.length > 2) {
+			const curKey = keys.shift();
+			if (!curKey) break; // should never happen
+			delete logs[section][curKey];
 		}
 	}
-	writeLogs();
-} catch (e) {
-	if (e.code !== 'ENOENT') throw e;
 }
+writeFile(LOGS_FILE, logs);
 
-function writeLogs() {
-	FS(LOGS_FILE).writeUpdate(() => (
-		JSON.stringify(logs)
-	));
+// Load bans
+
+bans = readFile(BANS_FILE);
+if (!bans) bans = {gameBans: {}, hostBans: {}};
+const now = Date.now();
+
+for (const userid in bans.gameBans) {
+	if (bans.gameBans[userid] < now) {
```

Comment:
Shouldnt it be <=?

Reply:
i don't think that literally a millisecond will make a difference